### PR TITLE
SRV_Channel: set default update rate only when necessary

### PR DESCRIPTION
**Pull request by rmackay9**
This resolves a problem in which Rover motors could pulse when the set-default-update-rate call was made by only making the call when we are actually changing the rate.  The pulse was caused by PWM outputs arriving very close together as shown below.

pulses-close-together

### CODE REVIEW

**Comment by rmackay9** [comment]
By the way, i've tested this and it resolves the motor jump issue.  I can still recreate the small motor jumps by setting the SERVO_RATE a few times to different rates but this is not something that users will do very often.

**Comment by rmackay9** [comment]
servo-rate-logic

Here's a picture of the servo output on a main output (servo0) and auxiliar (servo9) at the moment the servo rate is changed from 50hz to 100hz.  So it seems to be working.

File: libraries/AP_HAL_PX4/RCOutput.cpp [hunk 1]
```diff
@@ -644,10 +644,24 @@ void PX4RCOutput::set_output_mode(enum output_mode mode)
 // set default output update rate
 void PX4RCOutput::set_default_rate(uint16_t rate_hz)
 {
-    ioctl(_pwm_fd, PWM_SERVO_SET_DEFAULT_UPDATE_RATE, rate_hz);
+    // we use an unsigned long to match the ioctl definition
+    unsigned curr_rate_hz = 0;
```
Comment (OXINARF) [inline libraries/AP_HAL_PX4/RCOutput.cpp]: The comment says we use an unsigned long but it is actually an unsigned int.

**Comment by WickedShell** [comment]
This fixes servo twitching on Plane for me :)

**Comment by tridge** [comment]
just discussed with Randy, I think its better to just remember the last set value in RCOutput object, and avoid the ioctl. The ioctl involves a round trip to the io microcontroller, which costs us time in the main thread

**Comment by WickedShell** [comment]
Is this needed for the Plane 3.8 release or am I only seeing that I need it because I'm running master instead?

**Comment by rmackay9** [comment]
@WickedShell,
The change the caused the twitch is very new.  just in the past couple of days I think.

I've reworked the change to reduce the number of ioctl calls (and thus improve performance).

**Comment by rmackay9** [comment]
servo-rate-logic-2
Re-tested post changes and it seems to work. I'll probably wait another hour and then merge this is there are no further comments/concerns.

**Pull request merged**